File: ToyInterpreter 2/src/Model/IStack.java
package Model;

import mobserver.StackObserver;
import mobserver.VarTableObserver;
import Model.Statement.Statement;

public interface IStack {
	public void push(Statement s);
	public Statement top();
	public void pop();
	public boolean isEmpty();
	public String toString();
	public void addObserver(StackObserver o);
}

File: ToyInterpreter/ToyInterpreter/Model/IStack.cs
using System;
using ToyInterpreter.Model.MStatement;
using ToyInterpreter.Model.MObserver;

namespace ToyInterpreter.Model
{
	public interface IStack
	{
		void push(Statement s);
		Statement top();
		void pop();
		bool isEmpty();
		String toString();
		void addObserver(StackObserver o);
	}
}

